### PR TITLE
[edpm_download_cache] Download packages and containers images only when service will be deployed on the EDPM node.

### PR DESCRIPTION
**Pull request by rebtoor**

### CODE REVIEW

File: roles/edpm_download_cache/tasks/container_images.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@
     - download_cache
 
 - name: Download images for edpm_iscsid role
+  when: '"nova-storage" in edpm_services'
```
Comment (fao89) [inline roles/edpm_download_cache/tasks/container_images.yml:12]: We should also take into account the `edpm_services_override`,
maybe we can have a variable that defaults to `edpm_services` and gets `edpm_services_override` value if it is defined

Comment (rebtoor) [reply roles/edpm_download_cache/tasks/container_images.yml:12]: @slagle FYI

Comment (fao89) [reply roles/edpm_download_cache/tasks/container_images.yml:12]: I think we can have something like:
```yaml
edpm_running_services: {{ edpm_services_override | default(edpm_services) }}
```
so if you want the "default" services you can get `edpm_services` and if you want what is being executed you can get `edpm_running_services`

Comment (jpodivin) [reply roles/edpm_download_cache/tasks/container_images.yml:12]: That would be a good idea. The overrides are meant to take priority after all.

Comment (rebtoor) [reply roles/edpm_download_cache/tasks/container_images.yml]: It makes sense, I've added `edpm_download_cache_running_services` variable.

File: roles/edpm_download_cache/tasks/packages.yml [hunk 2]
```diff
@@ -57,6 +70,7 @@
     - download_cache
 
 - name: Download images for edpm_libvirt role
```
Comment (jpodivin) [inline roles/edpm_download_cache/tasks/packages.yml:72]: Not actually an image download. The role is using `dnf` to download packages. [0]

[0]https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_libvirt/tasks/download_cache.yml

**Review by jpodivin — CHANGES_REQUESTED** [review]

File: roles/edpm_download_cache/tasks/packages.yml (hunk 2 above)
Comment (rebtoor) [reply roles/edpm_download_cache/tasks/packages.yml:72]: oops, copy/paste gone wrong. Thanks Jiri.

**Comment by slagle** [comment]
recheck

**Review by fao89 — APPROVED** [review]

**Review by jpodivin — APPROVED** [review]

**Pull request merged**